Tasks/AzureTestPlanV0/automatedTests.ts

```diff
@@ -1,28 +1,38 @@
 import tl = require('azure-pipelines-task-lib/task');
-import { testInvoker } from './automatedTestInvoker'
-import { TestPlanData } from './testPlanData'
-import { publishAutomatedTestResult } from './publishAutomatedTests'
+import { testInvoker } from './automatedTestInvoker';
+import { TestPlanData } from './testPlanData';
+import { publishAutomatedTestResult } from './publishAutomatedTests';
 
+export async function automatedTestsFlow(testPlanInfo: TestPlanData, testSelectorInput: string): Promise<number> {
+  let listOfTestsToBeExecuted: string[] = testPlanInfo.listOfFQNOfTestCases;
+  let testInvokerStatusCode = 0;
 
-export async function automatedTestsFlow(testPlanInfo: TestPlanData, testSelectorInput: string) {
+  if (listOfTestsToBeExecuted !== null && listOfTestsToBeExecuted !== undefined && listOfTestsToBeExecuted.length > 0) {
+    tl.debug('Invoking test execution for tests: ' + listOfTestsToBeExecuted);
 
-    let listOfTestsToBeExecuted: string[] = testPlanInfo.listOfFQNOfTestCases;
-
-    console.log(tl.loc('automatedTestTriggered'));
-
-    if (listOfTestsToBeExecuted !== null && listOfTestsToBeExecuted !== undefined && listOfTestsToBeExecuted.length > 0) {
-        tl.debug("Invoking test execution for tests: " + listOfTestsToBeExecuted);
-        await testInvoker(listOfTestsToBeExecuted);
-        publishAutomatedTestResult(JSON.stringify(testPlanInfo.listOfAutomatedTestPoints));
+    try {
+      testInvokerStatusCode = await testInvoker(listOfTestsToBeExecuted);
+    } catch (err) {
+      tl.debug(`Unable to invoke automated test execution. Err:( ${err} )`);
     }
-    else {
-        console.log("No automated tests found for given test plan inputs ");
-        if (testSelectorInput === 'automatedTests') {
-            tl.setResult(tl.TaskResult.Failed, tl.loc('ErrorFailTaskOnNoAutomatedTestsFound'));
-        }
-        else {
-            tl.setResult(tl.TaskResult.Succeeded, "Successfully triggered manual test execution");
-        }
+
+    try {
+      await publishAutomatedTestResult(JSON.stringify(testPlanInfo.listOfAutomatedTestPoints));
+    } catch (err) {
+      tl.error(`Error while publishing automated Test Results with err : ( ${err} )`);
+      return 1;
```

Comment:
Just for my clarification, in case of publish failures are we setting the task result as failed or will it depend on execution only?

Reply:
Both. We are propagating the status code through the lifecycle.